apps/near_app/near_pub_key.c

```diff
@@ -0,0 +1,444 @@
+                             const char *wallet_name) {
+  char msg[100] = "";
+
+  if (NEAR_QUERY_GET_PUBLIC_KEYS_TAG == which_request) {
+    snprintf(msg,
+             sizeof(msg),
+             UI_TEXT_ADD_ACCOUNT_PROMPT,
+             NEAR_CHAIN_NAME,
+             wallet_name);
+  } else {
+    snprintf(msg,
+             sizeof(msg),
+             UI_TEXT_RECEIVE_TOKEN_PROMPT,
+             NEAR_TOKEN_NAME,
+             NEAR_CHAIN_NAME,
+             wallet_name);
+  }
+
+  return core_scroll_page(NULL, msg, near_send_error);
+}
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+void near_get_pub_keys(near_query_t *query) {
+  char wallet_name[NAME_SIZE] = "";
+  uint8_t seed[64] = {0};
+
+  const pb_size_t which_request = query->which_request;
+  const near_get_public_key_intiate_request_t *init_req = NULL;
+  pb_size_t which_response;
+
+  if (NEAR_QUERY_GET_PUBLIC_KEYS_TAG == which_request) {
+    which_response = NEAR_RESULT_GET_PUBLIC_KEYS_TAG;
+    init_req = &query->get_public_keys.initiate;
+  } else {
+    which_response = NEAR_RESULT_GET_PUBLIC_KEY_TAG;
+    init_req = &query->get_public_key.initiate;
+  }
+
+  const pb_size_t count = init_req->derivation_paths_count;
+
+  uint8_t pubkey_list[sizeof(init_req->derivation_paths) /
+                      sizeof(near_get_public_key_derivation_path_t)]
+                     [NEAR_PUB_KEY_SIZE] = {0};
+
+  // TODO: Handle wallet search failures - eg: Wallet ID not found, Wallet ID
+  // found but is invalid/locked wallet
+  if (!check_which_request(query, NEAR_GET_PUBLIC_KEY_REQUEST_INITIATE_TAG) ||
+      !validate_request(init_req, which_request) ||
+      !get_wallet_name_by_id(init_req->wallet_id, (uint8_t *)wallet_name)) {
+    return;
+  }
+
+  // Take user consent to export public key for the wallet
+  if (!get_user_consent(which_request, wallet_name)) {
+    return;
+  }
+
+  core_status_set_flow_status(NEAR_GET_PUBLIC_KEY_STATUS_CONFIRM);
+
+  // TODO: Handle rejections during wallet reconstruction flows - eg: cancel
+  // button pressed on PIN/Passphrase entry
+  if (!reconstruct_seed_flow(query->get_public_key.initiate.wallet_id,
+                             &seed[0])) {
+    // TODO: Handle error case reporting to host
+    return;
+  }
+
+  core_status_set_flow_status(NEAR_GET_PUBLIC_KEY_STATUS_SEED_GENERATED);
+
+  delay_scr_init(ui_text_processing, DELAY_SHORT);
+
+  // Fill the uncompressed public keys and ASCII formatted address into local
+  // buffers
```

Comment:
we are not filling ascii formatted address. please update

Reply:
Updated